### PR TITLE
Fix zero Edisp, add replacement tally

### PR DESCRIPTION
**Pull request by dschwen**
Also adds _Quick calculation of damage_ (Kinchin-Pease).

ACloses #62

### CODE REVIEW

File: apps/src/TrimRange.C [hunk 1]
```diff
@@ -10,6 +10,13 @@ TrimRange::TrimRange(SimconfType * simconf, SampleBase * sample) :
 }
 
 void
+TrimRange::vacancyCreation()
+{
+  // Quick calculation of Damage
+  _simconf->vacancies_created += _recoil->_E * 0.4 / _element->_Edisp;
```
Comment (snschune) [inline apps/src/TrimRange.C]: we should actually do this
http://www.iue.tuwien.ac.at/phd/hoessinger/node47.html

**Review by snschune — CHANGES_REQUESTED** [review]

File: apps/src/TrimVacCount.C [hunk 2]
```diff
@@ -22,6 +22,16 @@ TrimVacCount::vacancyCreation()
 }
 
 void
+TrimVacCount::replacementCollision()
+{
+  int x = _recoil->_pos(0);
```
Comment (snschune) [inline apps/src/TrimVacCount.C:27]: What is `_recoil->_pos(0)`. Is that the x-coordinate?

File: apps/src/TrimVacCount.C [hunk 3]
```diff
@@ -30,13 +40,19 @@ TrimVacCount::threadJoin(const ThreadedTrimBase & ttb)
...
+  for (unsigned int x = 0; x < tvc._repl_bin.size(); ++x)
+    _repl_bin[x] += tvc._repl_bin[x];
 }
 
 void
 TrimVacCount::writeOutput()
 {
   // write out vaccancy histogram
+  const unsigned int size = std::max(_vac_bin.size(), _repl_bin.size());
+  _vac_bin.resize(size);
```
Comment (snschune) [inline apps/src/TrimVacCount.C:53]: this just adds zeros at the end if the other <>_bin is larger?

File: trim.C [hunk 4]
```diff
@@ -355,6 +355,7 @@ TrimBase::trim(IonBase * pka, std::queue<IonBase*> & recoils)
           vacancyCreation();
         } else {
           // nope, the _pka gets stuck at that site as...
+          replacementCollision();
```
Comment (snschune) [inline trim.C:358]: replacement in this sense now includes replacement & substitution. I guess that is what we want.

File: apps/src/TrimRange.C (hunk 1 above)
Comment (dschwen) [reply apps/src/TrimRange.C]: Yes, _modified_ KP. I'll implement that.

File: apps/src/TrimVacCount.C (hunk 2 above)
Comment (dschwen) [reply apps/src/TrimVacCount.C:27]: Yes

File: apps/src/TrimVacCount.C (hunk 3 above)
Comment (dschwen) [reply apps/src/TrimVacCount.C:53]: Yes

File: trim.C (hunk 4 above)
Comment (dschwen) [reply trim.C:358]: Yes. The distinction can still be made in the callback if necessary.

File: apps/src/TrimVacCount.C (hunk 2 above)
Comment (snschune) [reply apps/src/TrimVacCount.C:27]: Should't this be divided by some bin width to map it into a regular grid index?

**Comment by dschwen** [comment]
Modified KP takes it from

```
Vacancies/ion: 2082.06 
```

to:

```
Vacancies/ion: 1536.79
```

for the 150keV Copper on Copper test. This is a lot closer to the SRIM quick damage result.

**Comment by snschune** [comment]
👍

File: apps/src/TrimVacCount.C (hunk 2 above)
Comment (snschune) [reply apps/src/TrimVacCount.C:27]: Still curious how this works exactly, but I'll merge it for now.

Comment (dschwen) [reply apps/src/TrimVacCount.C:27]: The bin index is the depth in angstroms

Comment (snschune) [reply apps/src/TrimVacCount.C:27]: Ok so you don't allow a different bin size but `delx = 1 A`.

**Pull request merged**